src/OpenLoco/src/Ui/WindowManager.cpp

```diff
@@ -2034,6 +2035,137 @@ namespace OpenLoco::Ui::WindowManager
     {
         return ((getVehiclePreviewRotationFrame() + 2) / 4) & 0x3F;
     }
+
+    static void windowDraw(Gfx::RenderTarget* rt, Ui::Window* w, Rect rect);
+    static void windowDraw(Gfx::RenderTarget* rt, Ui::Window* w, int16_t left, int16_t top, int16_t right, int16_t bottom);
+    static bool windowDrawSplit(Gfx::RenderTarget* rt, Ui::Window* w, int16_t left, int16_t top, int16_t right, int16_t bottom);
+
+    /**
+     * 0x004C5EA9
+     *
+     * @param w
+     * @param left @<ax>
+     * @param top @<bx>
+     * @param right @<dx>
+     * @param bottom @<bp>
+     */
+    static void windowDraw(Gfx::RenderTarget* rt, Ui::Window* w, int16_t left, int16_t top, int16_t right, int16_t bottom)
+    {
+        if (!w->isVisible())
+            return;
+
+        // Split window into only the regions that require drawing
+        if (windowDrawSplit(rt, w, left, top, right, bottom))
+            return;
+
+        // Clamp region
+        left = std::max(left, w->x);
+        top = std::max(top, w->y);
+        right = std::min<int16_t>(right, w->x + w->width);
+        bottom = std::min<int16_t>(bottom, w->y + w->height);
+        if (left >= right)
+            return;
+        if (top >= bottom)
+            return;
+
+        // Draw the window in this region
+        Ui::WindowManager::drawSingle(rt, w, left, top, right, bottom);
+
+        for (uint32_t index = Ui::WindowManager::indexOf(w) + 1; index < Ui::WindowManager::count(); index++)
```

Comment:
As this now resides in the WindowManager namespace, this can be simplified to remove the `Ui::WindowManager`` prefixes, right?

Reply:
You are right, ill address that. Small oversight from just copying it over.